### PR TITLE
Rework of the FEM API

### PR DESCRIPTION
**Pull request by czeslawmakarski**
First step to rework FEM API, code still buggy.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 *.orig
 .deps/
 .dirstamp
+.vscode/
```
Comment (czeslawmakarski) [inline .gitignore]: Is it OK to merge such changes upstream?

File: src/nrf_802154_core.c [hunk 2]
```diff
@@ -781,75 +786,81 @@ static void fem_for_lna_set(nrf_timer_cc_channel_t cc_channel,
...
+        timer.event.timer.p_timer_instance = NRF_802154_TIMER_INSTANCE;
+        timer.event.timer.compare_channel = NRF_TIMER_CC_CHANNEL0;
+        timer.event.timer.counter_value = NRF_FEM_RADIO_RX_STARTUP_LATENCY_US;
+
+        radio.type = NRF_802154_PA_LNA_EVENT_TYPE_RADIO;
+        radio.ignore_ppi = true;
+        radio.ppi_ch_id = PPI_CCAIDLE_FEM;
+        radio.event.radio.event_register = (uint32_t)nrf_radio_event_address_get(NRF_RADIO_EVENT_CCAIDLE);
+
+        ret = ((nrf_802154_pa_lna_lna_configuration_set(&timer, &radio) == NRFX_SUCCESS) &&
```
Comment (czeslawmakarski) [inline src/nrf_802154_core.c]: Use `NRF_` exit codes instead of `NRFX`.

File: src/fem/nrf_fem_control.c [hunk 3]
```diff
@@ -81,320 +58,250 @@ static bool pin_is_enabled(nrf_fem_control_pin_t pin)
...
-    *p_cfg = m_nrf_fem_control_cfg;
-}
-
-void nrf_fem_control_activate(void)
-{
-    if (m_nrf_fem_control_cfg.pa_cfg.enable || m_nrf_fem_control_cfg.lna_cfg.enable)
+    else
     {
-        nrf_ppi_channel_enable((nrf_ppi_channel_t)m_nrf_fem_control_cfg.ppi_ch_id_clr);
+        chan = (activate /* ^ p_pin_config->active_high */) ? m_nrf_fem_interface_config.ppi_set_ch_id_set : m_nrf_fem_interface_config.ppi_ch_id_clr;
```
Comment (czeslawmakarski) [inline src/fem/nrf_fem_control.c]: Debug comment.

File: src/fem/nrf_fem_control.c [hunk 4]
```diff
@@ -81,320 +58,250 @@ static bool pin_is_enabled(nrf_fem_control_pin_t pin)
...
-
-            case NRF_FEM_CONTROL_PA_PIN:
-                target_time = NRF_FEM_RADIO_TX_STARTUP_LATENCY_US - NRF_FEM_PA_TIME_IN_ADVANCE;
-                break;
-
-            default:
-                assert(false);
-                break;
-        }
+        return -1;
```
Comment (czeslawmakarski) [inline src/fem/nrf_fem_control.c]: Use some meaningful return code.

**Comment by hubertmis** [comment]
Should we create a temporary integration branch to merge this feature while it's buggy? Or do you want us to just review and not merge for now?

**Comment by czeslawmakarski** [comment]
> 
> 
> Should we create a temporary integration branch to merge this feature while it's buggy? Or do you want us to just review and not merge for now?

I was thinking about the second option, but maybe the first one is also viable.

**Comment by hubertmis** [comment]
Let's stick to just review for now. We'll decide later when we want to merge it and which destination branch should we use.

**Pull request closed without merging**